addons/website_sale/static/tests/tours/website_sale_shop_editor_tour.js

```diff
@@ -18,3 +18,26 @@ wTourUtils.registerWebsitePreviewTour("shop_editor", {
     extra_trigger: "iframe div.o_pricelist_dropdown a[data-bs-toggle=dropdown][aria-expanded=false]",
     content: "Click on the pricelist again.",
 }]);
+
+wTourUtils.registerWebsitePreviewTour("shop_editor_set_product_ribbon", {
+    test: true,
+    url: "/shop",
+    edition: true,
+}, [{
+    content: "Click on first product",
+    trigger: "iframe .oe_product:first",
+}, {
+    content: "Open the ribbon selector",
+    trigger: ".o_wsale_ribbon_select we-toggler",
+    extra_trigger: ".o_wsale_ribbon_select",
+}, {
+    content: "Select a ribbon",
+    trigger: 'we-button:contains("Sale")',
+}, {
+    content: "Save",
+    trigger: 'button[data-action="save"]',
+}, {
+    content: "Close the editor",
+    trigger: 'iframe .oe_product:first .o_ribbon:contains("Sale")',
+    extra_trigger: 'iframe body:not(.editor_enable)',
```

Comment:
- The "content" does not seem to describe what you are doing here. Something like "Check that the ribbon was properly saved" maybe?
- Since this is a check and the last step is normally not needed to add something like `run: () => null` but I would add it anyway so that you can remember that in 16.4 you would have to write `isCheck: true` instead.
- Since you check for `iframe body:not(.editor_enable)` to know when you left the edit mode, I would add an extra trigger in your first step: `body.editor_enable`. Otherwise it means that if we rename `editor_enable` in the future and forget about this file, this tour would not test nothing since `iframe body:not(.editor_enable)` would always be true.

Reply:
@qsm-odoo thank you for the comments :pray: ! I applied the suggested changes